src/components/TodoForm.js

```diff
@@ -1,106 +1,177 @@
 import { useEffect, useState } from "react";
 import useTodos from "../hooks/useTodos";
 import { v4 as uuidv4 } from "uuid";
-import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { faCalendarAlt } from "@fortawesome/free-solid-svg-icons";
-import DatePicker from "react-datepicker";
-import "react-datepicker/dist/react-datepicker.css";
-import dayjs from "dayjs";
 
 const TodoForm = () => {
   const { addTodo, editingTodo, isEdit } = useTodos();
 
   const defaultValues = {
     id: "",
-    task: "",
+    title: "",
     description: "",
+    completed: false,
     note: "",
     date: "",
-    completed: false,
+    priority: "",
   };
 
   const [task, setTask] = useState(defaultValues);
-  const [showNoteInput, setShowNoteInput] = useState(false);
-  const [showDateInput, setShowDateInput] = useState(false);
+  const [isTaskInputVisible, setIsTaskInputVisible] = useState(false);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalContent, setModalContent] = useState("");
+  const [isPriorityDropdownVisible, setIsPriorityDropdownVisible] =
+    useState(false);
 
   useEffect(() => {
     if (isEdit) {
       setTask(editingTodo);
+      setIsTaskInputVisible(true);
     }
   }, [editingTodo, isEdit]);
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    let id = uuidv4();
-    if (isEdit) {
-      id = editingTodo.id;
-    }
+    const id = isEdit ? editingTodo.id : uuidv4();
     addTodo({ ...task, id });
     setTask(defaultValues);
-    setShowNoteInput(false);
-    setShowDateInput(false);
+    setIsTaskInputVisible(false);
+  };
+
+  const openModal = (content) => {
+    setModalContent(content);
+    setIsModalOpen(true);
+  };
+
+  const closeModal = () => {
+    setIsModalOpen(false);
   };
 
-  const handleAddNoteClick = () => {
-    setShowNoteInput(!showNoteInput);
+  const handleModalSubmit = (e) => {
```

Comment:
forma gerek yok gibi burada state tut yeterli

Reply:
anlamadim abi

Reply:
Modal icin form tutuyorsun. Ama bu gönderilecek bir form değil, dıştaki form için bu bilgileri alıyorsun. Bu degerleri state olarak tutmak yeterli, formluk isin yok